### PR TITLE
Retry support

### LINKED ISSUE #10: Conflict Retries
It's possible that when many controllers/goroutines are trying to make updates to aws-auth using this library, that some of them will fail due to conflict of resourceVersion.

We should have an option to retry with exponential backoff when this happens.

Example
```
$ aws-auth upsert --username test3 --mapusers --userarn role2 --retry
```

### PR DESCRIPTION
**Pull request by eytan-avisror**
Fixes #10 

Adds retry support for updating configmaps in cases of conflicts

### CODE REVIEW

File: pkg/mapper/remove.go [hunk 1]
```diff
@@ -24,8 +24,6 @@ import (
 
 // Remove removes by match of provided arguments
 func (b *AuthMapper) Remove(args *RemoveArguments) error {
-	args.Validate()
```
Comment (nvandanapu) [inline pkg/mapper/remove.go:27]: I think just having a client-side validation is not enough. Do you think we can bring this back here?

File: pkg/mapper/upsert.go [hunk 2]
```diff
@@ -22,8 +22,6 @@ import (
 
 // Upsert update or inserts by rolearn
 func (b *AuthMapper) Upsert(args *UpsertArguments) error {
-	args.Validate()
```
Comment (nvandanapu) [inline pkg/mapper/upsert.go:25]: Ditto. Can you let me know why this validation is moved to cli client?

File: pkg/mapper/remove.go (hunk 1 above)
Comment (eytan-avisror) [reply pkg/mapper/remove.go:27]: Good point, I will move those back

Comment (eytan-avisror) [reply pkg/mapper/remove.go:27]: fixed

**Pull request merged**